test/e2e/instrumentation/events.go

```diff
@@ -0,0 +1,182 @@
+		Note:                "This is " + name,
+		Reason:              "Test",
+		Type:                "Normal",
+		ReportingController: "test-controller",
+		ReportingInstance:   "test-node",
+	}
+}
+
+func listEvents(name, namespace string, client eventsv1beta1.EventInterface) bool {
+	eventsList, err := client.List(context.TODO(), metav1.ListOptions{
+		LabelSelector: "testevent-constant=true",
+	})
+	framework.ExpectNoError(err, "failed list all events")
+
+	foundCreatedEvent := false
+	for _, val := range eventsList.Items {
+		if val.ObjectMeta.Name == name && val.ObjectMeta.Namespace == namespace {
+			foundCreatedEvent = true
+			break
+		}
+	}
+	return foundCreatedEvent
+}
+
+var _ = common.SIGDescribe("Events API", func() {
+	f := framework.NewDefaultFramework("events")
+	client := f.ClientSet.EventsV1beta1().Events(f.Namespace.Name)
+	clientAllNamespaces := f.ClientSet.EventsV1beta1().Events(metav1.NamespaceAll)
+
+	/*
+		Release : v1.19
+		Testname: Event resource lifecycle
+		Description: Create an event, the event MUST exist.
+		The event is patched with a new note, the check MUST have the update note.
+		The event is deleted and MUST NOT show up when listing all events.
+	*/
+	ginkgo.It("should ensure that an event can be fetched, patched, deleted, and listed", func() {
+		eventTestName := "event-test"
+
+		ginkgo.By("creating a test event")
+		// create a test event in test namespace
+		_, err := client.Create(context.TODO(), testEvent(eventTestName, "testevent-constant"), metav1.CreateOptions{})
+		framework.ExpectNoError(err, "failed to create test event")
+
+		ginkgo.By("listing all events in all namespaces")
+		// get a list of Events in all namespaces to ensure endpoint coverage
+		foundCreatedEvent := listEvents(eventTestName, f.Namespace.Name, clientAllNamespaces)
+		framework.ExpectEqual(foundCreatedEvent, true, "unable to find the test event")
+
+		ginkgo.By("listing all events in framework namespace")
+		// get a list of Events in framework namespace to ensure endpoint coverage
+		foundCreatedEvent = listEvents(eventTestName, f.Namespace.Name, client)
+		framework.ExpectEqual(foundCreatedEvent, true, "unable to find the test event")
+
+		ginkgo.By("patching the test event")
+		// patch the event's note
+		eventPatchNote := "This is a test event - patched"
+		eventPatch, err := json.Marshal(map[string]interface{}{
+			"note": eventPatchNote,
+		})
+		framework.ExpectNoError(err, "failed to marshal the patch JSON payload")
+
+		_, err = client.Patch(context.TODO(), eventTestName, types.StrategicMergePatchType, []byte(eventPatch), metav1.PatchOptions{})
+		framework.ExpectNoError(err, "failed to patch the test event")
+
+		ginkgo.By("getting the test event")
+		// get event by name
+		event, err := client.Get(context.TODO(), eventTestName, metav1.GetOptions{})
+		framework.ExpectNoError(err, "failed to fetch the test event")
+		framework.ExpectEqual(event.Note, eventPatchNote, "test event note does not match patch note")
```

Comment:
Sorry - I should have been more clear. Let's make something like:
```
// at the beginning of the test
testEvent := newTestEvent(...)
...
// here
testEvent.Note = eventPatchNote
// clear ResourceVersion field which is on API calls
event.ObjectMeta.ResourceVersion = ""
if !apiequality.Semantic.DeepEqual(testEvent, event) {
  framework.Failf("test event wasn't properly patched: %v", diff.ObjectReflectDiff(testEvent, event)
}
```

[diff is from "k8s.io/apimachinery/pkg/util/diff" ]
```

Reply:
Done.